server.js

```diff
@@ -55,23 +55,50 @@ app.get('/blocks/:index', (request, response) => {
 
 app.post('/mining/submit-mined-block', (request, response) => {
    let minedRes = burgerNode.addMinedBlock(request.body);
-    burgerSync.broadcastNewBlock(burgerNode.chain);
-    if(minedRes[0]){
+    burgerSync.broadcastNewBlock(request.body);
+
+    const minedBlockAccepted = minedRes[0];
+    const result = minedRes[1];
+    const resultType = minedRes[2];
+
+    const minedResultType = burgerNode.chain.resultType;
+    switch (resultType) {
+      case minedResultType.VALID_BLOCK:
         response.json({
-            "message":minedRes[1]
+            "message": result
         }).status(200);
-    }else{
-        if(minedRes[1]==="Block not found or already mined"){
-            response.json({
-                "errorMsg":minedRes[1]
-            }).status(404);
-        }else{
-            response.json({
-                "errorMsg":minedRes[1]
-            }).status(404);
-        }
+        break;
+      case minedResultType.INVALID_BLOCK:
+        response.json({
+            "errorMsg": result
+        }).status(404);
```

Comment:
The `status(404)` must go first before `.json()` like this:
```
response.status(404).json({
            "errorMsg": result
        });
```

Or else we'll receive a 200 even though it failed.

Reply:
ah, this wasn't my code, I just copied and pasted old code, do you want me to change it?

Reply:
Yes, please.

Reply:
i'm busy with dApp stuff now :(

Reply:
May I change it? :D